### PR TITLE
Fix decimal precision loss in cast: float->decimal

### PR DESCRIPTION
**Pull request by Matt711**
## Description

`cudf::cast` from float to decimal truncates instead of rounds. When translating the polars plan, we convert decimal `TrueDivide` operations to float division then cast back to decimal (since libcudf doesn't support `TrueDivide` between decimals, see #20077). But since the cast truncates, we have to cast to a higher precision decimal first, then round.

contributes to #19939 
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf_polars/cudf_polars/containers/column.py [hunk 1]
```diff
@@ -337,6 +344,29 @@ def astype(self, dtype: DataType, stream: Stream, *, strict: bool = True) -> Col
...
+                    plc.unary.cast(
+                        self.obj,
+                        plc.DataType(
+                            plc.TypeId.DECIMAL128,
+                            scale=-max(target_scale, _FLOAT64_DECIMAL_PRECISION),
+                        ),
+                        stream=stream,
+                    ),
+                    decimal_places=target_scale,
+                    round_method=plc.round.RoundingMethod.HALF_EVEN,
```
Comment (bdice) [inline python/cudf_polars/cudf_polars/containers/column.py:371]: Just checking: polars uses “half even” rounding? I thought that mode was pretty uncommon across engines. A reference would be appropriate if one exists.

File: python/cudf_polars/tests/expressions/test_casting.py [hunk 2]
```diff
@@ -87,3 +89,15 @@ def test_cast_to_string_unsupported():
     df = pl.LazyFrame({"a": [True]})
     query = df.select(pl.col("a").cast(pl.String()))
     assert_ir_translation_raises(query, NotImplementedError)
+
+
+def test_float_to_decimal_rounding():
```
Comment (bdice) [inline python/cudf_polars/tests/expressions/test_casting.py:95]: Can we add explicit tests for the half-even rounding mode?

File: python/cudf_polars/cudf_polars/containers/column.py (hunk 1 above)
Comment (Matt711) [reply python/cudf_polars/cudf_polars/containers/column.py:371]: > Just checking: polars uses “half even” rounding? I thought that mode was pretty uncommon across engines. 

Banker's rounding is default

> A reference would be appropriate if one exists.

Sure

File: python/cudf_polars/tests/expressions/test_casting.py (hunk 2 above)
Comment (Matt711) [reply python/cudf_polars/tests/expressions/test_casting.py:95]: There are already tests for it

Comment (Matt711) [reply python/cudf_polars/tests/expressions/test_casting.py:95]: https://github.com/rapidsai/cudf/blob/93fa04bb2681906227d474656ca2f3ed4fac001e/python/cudf_polars/tests/expressions/test_numeric_unaryops.py#L167

**Comment by Matt711** [comment]
> I think this makes sense. One question: is there an edge case we aren't handling when the target scale is already high? But IIUC, that might be unavoidable until libcudf supports decimals in true division?

The truncation vs. rounding doesn't really matter once the scale is high enough because this is somewhat specific to `Float64`. As far as true division support in libcudf, there's no plan to add support in libcudf AFAIK.
